packages/ffe-checkbox-react/src/Checkbox.js

```diff
@@ -1,20 +1,28 @@
 import React from 'react';
-import { bool, node, string } from 'prop-types';
+import { bool, node, string, func } from 'prop-types';
 import { v4 as hash } from 'uuid';
 import classNames from 'classnames';
 
 export default function CheckBox(props) {
-    const { children, inline, invalid, label, noMargins, ...rest } = props;
+    const { children, inline, invalid, label, noMargins, onClick, stopPropagation, ...rest } = props;
 
     const id = props.id || `checkbox-${hash()}`;
 
+    const onClickInput = event => {
```

Comment:
If this had worked, it would've been great. I mentioned in the issue that I've tried this, but I can explain a bit closer here.
When you click on the checkbox or the label, you actually get a click-event bubbling from tag name `input` (I'm catching this with `onClick` in a surrouning `span`).
When I try to stop it (regardless of target tag name):
```javascript
<Checkbox onClick={ event => event.stopPropagation() } />
```
I still get a bubbling click-event. The only difference is that now it has target tag name `label`.
I'm not sure exactly how the events spawn and bubble when you have a label and an input, but so far the only way I've found of stopping it is to stop propagation from both the `input` and the `label`.

If you have any other suggestions, I'd be happy to explore them!

Reply:
is `stopImmediatePropagation()` doing better work?

Reply:
> When I try to stop it (regardless of target tag name) I still get a bubbling click-event. The only difference is that now it has target tag name label.

The tag name is changed when you stop propagation? Wat.
Maybe [stopImmediatePropagation](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopImmediatePropagation) like Ivar suggests would work? 

> If several listeners are attached to the same element for the same event type, they are called in order in which they have been added. If during one such call, event.stopImmediatePropagation() is called, no remaining listeners will be called.

The behavior you describe could be down to the two elements being involved. On the other hand it may be two sources, not two listeners so YMMV.

If none of this works I think you should roll your own version of `Checkbox` to be honest. It's not a very complex component and I don't like this edge case API being there for all consumers, the vast majority of which won't need this.

Reply:
`Uncaught TypeError: event.stopImmediatePropagation is not a function`
Does that happen if there's only one listener?

I see I misinterpreted the events a bit before. I dug a bit more and see more clearly what happens now.

When you click on the label, you actually get two click-events bubbling. One from the `input` and one from the `label`. This is why it seemed like the expandable row I mentioned in the issue didn't expand to begin with. The truth is it both expanded and collapsed.
Focusing the label and pressing space (to toggle the box) only triggers a click event from the `input`, and thus the row is expanded/collapsed.

When you add
```javascript
<Checkbox onClick={ event => event.stopPropagation() } />
```
you do in fact stop the click event from the `input`. So now focusing and using spacebar is fine - there's no event bubbling. However, the event from the `label` is still bubbling, and thus the row is expanded/collapsed.

It still seems to me that you need to explicitly stop propagation of both the `label` and the `input` click events.

Reply:
Right, it's probably a synthetic React-event and not a native event so it most likely doesn't have stopImmediate as part of its API.

Reply:
> `event.nativeEvent.stopImmediatePropagation()` should not produce the error.

Thanks for the clearification. This does indeed not produce the error, but it doesn't help stopping the click event bubbling from the `label`.